docker/build_moab.sh

```diff
@@ -1,34 +1,39 @@
 #!/bin/bash
 
-# $1: compiler (gcc-4.8, gcc-5, gcc-6, gcc-7, clang-4.0, clang-5.0)
-# $2: moab version (5.1.0, master)
+# $1: compiler (gcc, clang)
+# $2: hdf5 version (system, 1.10.4)
+# $3: moab version (5.1.0, master)
 
 set -e
 
-source /root/etc/$1.env
-moab_version=$2
-hdf5_version=1.8.13
+export compiler=$1
+export hdf5_version=$2
+export moab_version=$3
+
+source /root/etc/env.sh
 
-mkdir -p ${build_dir}/moab-${moab_version}/bld
-rm -rf ${install_dir}/moab-${moab_version}
-cd ${build_dir}/moab-${moab_version}
 if [[ ${moab_version} == "master" ]]; then
-  git clone https://bitbucket.org/fathomteam/moab -b ${moab_version}
+  branch=${moab_version}
 else
-  git clone https://bitbucket.org/fathomteam/moab -b Version${moab_version}
+  branch=Version${moab_version}
 fi
+
+rm -rf ${moab_build_dir}/bld ${moab_install_dir}
+mkdir -p ${moab_build_dir}/bld
+cd ${moab_build_dir}
+git clone https://bitbucket.org/fathomteam/moab -b ${branch}
 ln -snf moab src
 cd moab
 autoreconf -fi
 cd ../bld
-../src/configure --enable-dagmc \
-                 --disable-ahf \
+../src/configure --enable-pymoab \
                  --enable-shared \
                  --enable-optimize \
                  --disable-debug \
-                 --with-hdf5=${install_dir}/hdf5-${hdf5_version} \
-                 --prefix=${install_dir}/moab-${moab_version} \
+                 --with-hdf5=${hdf5_install_dir} \
+                 --prefix=${moab_install_dir} \
```

Comment:
Because then you have to build it twice to get the shared and static libraries. CMake can't build them at the same time, while autotools can.

That being said, it might be a good idea to switch to CMake always though, since I get the feeling the MOAB CMake build is maintained much more than the autotools build.